ui/components/app/multi-rpc-edit-modal/network-list-item/network-list-item.test.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { Provider } from 'react-redux';
+import NetworkListItem from './network-list-item'; // Adjust the path as needed
+import '@testing-library/jest-dom/extend-expect';
+
+const mockStore = configureMockStore([thunk]);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const setup = (props: any) => {
+  const store = mockStore({});
+  return render(
+    <Provider store={store}>
+      <NetworkListItem {...props} />
+    </Provider>,
+  );
+};
+
+describe('NetworkListItem', () => {
+  it('renders correctly with required props', () => {
+    const { container } = setup({
+      networkConfiguration: {
+        chainId: '0x1',
+        name: 'Ethereum Mainnet',
+        defaultRpcEndpointIndex: 0,
+        rpcEndpoints: [
+          {
+            name: 'Infura Mainnet',
+            url: 'https://mainnet.infura.io/v3/YOUR_INFURA_KEY',
```

Comment:
do these rpcEndpoints need the `type` property?

Reply:
https://github.com/MetaMask/metamask-extension/pull/26433/commits/838529c5d346a0cd1e3b0e4bc2fdb5fa76a90b40

Reply:
https://github.com/MetaMask/metamask-extension/pull/26433/commits/702a3e2fba9f54a454c6d643d979f6fbea11cc37